src/lib.rs

```diff
@@ -25,6 +35,8 @@ pub enum ApiError {
     /// There was an error parsing a URL from a string.
     #[error("Error parsing URL: {0}")]
     UrlParseError(#[from] url::ParseError),
+    #[error("Dates must be after 2018-05-10")]
+    DateTooOld,
```

Comment:
Maybe an error is too drastic and we should keep it simple and user friendly: if the start date is before OLDEST_VALID_DATE we simply use the latter without telling the user - it is an optimisation in a way. If the end date is before OLDEST_VALID_DATE we could return no results and the user doesn't have to handle the error. What do you think?

Reply:
Done [here](https://github.com/jnioche/carbonintensity-api/pull/20/commits/477ff46182455d9e97d02e75a40eb41c0644c386) and [here](https://github.com/jnioche/carbonintensity-api/pull/20/commits/843c4551387ea4671bf1d043160a6245d07ce667).

I think it's nicer. And I don't see much of a drawback of defaulting to the oldest valid date in this case.